Providers/Scripts/2.6x-2.7x/Scripts/Tests/test_nxProviders.py

```diff
@@ -3579,6 +3580,107 @@ def testGetOMSCustomLog_del(self):
         print 'GET '+ repr(g)
         self.assertTrue(check_values(g, m)  ==  True, \
         'Get('+repr(g)+' should return ==['+repr(m)+']')
+
+
+@unittest2.skipUnless(os.system('ps -ef | grep -v grep | grep omsagent') ==
+                      0,'Skipping nxOMSCustomLogNewTestCases.   omsagent is not running.')
+class nxOMSCustomLogNewTestCases(unittest2.TestCase):
+    """
+    Test Case for nxOMSCustomLogNew.py
+    """
+
+    original_conf_path = None
+    mock_conf_path = './ut_customlognew.conf'
+
+    def setUp(self):
+        """
+        Setup test resources
+        """
+        self.original_conf_path = nxOMSCustomLogNew.CONF_PATH
+        nxOMSCustomLogNew.CONF_PATH = self.mock_conf_path
+        os.system('rm -rf {0}'.format(self.mock_conf_path))
+        
+        nxOMSCustomLogNew.RESOURCE_DIR = '/var/tmp/omsconfig/modules/nxOMSCustomLogNew/'
+        nxOMSCustomLogNew.RESOURCE_PLUGIN_DIR = nxOMSCustomLogNew.RESOURCE_DIR + 'Plugin/'
+        nxOMSCustomLogNew.PLUGIN = nxOMSCustomLogNew.RESOURCE_PLUGIN_DIR + 'in_sudo_tail.rb'
+        nxOMSCustomLogNew.SCRIPT = nxOMSCustomLogNew.RESOURCE_PLUGIN_DIR + 'tailfilereader.rb'
+        nxOMSCustomLogNew.AGENT_PLUGIN_DIR = '/var/tmp/omsagent/plugin/'
+      
+        os.system('rm -rf /var/tmp/omsconfig /var/tmp/omsagent;') 
+
+        os.system('mkdir -p ' + nxOMSCustomLogNew.RESOURCE_PLUGIN_DIR + ';' 
+              'mkdir -p ' + nxOMSCustomLogNew.AGENT_PLUGIN_DIR + ';' 
+              'echo sudo_tail plugin >> ' + nxOMSCustomLogNew.PLUGIN + ';' 
+              'echo tailfilereader script >> ' + nxOMSCustomLogNew.SCRIPT + ';'
+        )
+
+    def tearDown(self):
+        """
+        Remove test resources
+        """
+        nxOMSCustomLogNew.CONF_PATH = self.original_conf_path
+        os.system('rm -rf /var/tmp/omsconfig /var/tmp/omsagent')
```

Comment:
Not sure if other parts of DSC/OMSAgent use these tmp folders, but you should only remove the explicit files this code creates.